### PR TITLE
[PM-10104] add generator password type policy

### PR DESCRIPTION
**Pull request by audreyality**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10104

## 📔 Objective

Replace "Default Type" policy with "Password Type" policy. 

This policy is consumed in #10277.

## 📸 Screenshots

Before - UI does not change based on password type selection:
image

After - UI displays only those options that match the selected type:

image

image

image

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/locales/en/messages.json [hunk 1]
```diff
@@ -4102,6 +4102,10 @@
   "defaultType": {
     "message": "Default type"
   },
```
Comment (djsmith85) [inline apps/web/src/locales/en/messages.json:4104]: `defaultType` should be removed, as the only usage was in `apps/web/src/app/admin-console/organizations/policies/password-generator.component.html`

Comment (audreyality) [reply apps/web/src/locales/en/messages.json]: Addressed in 5895751905d5e.

**Review by djsmith85 — APPROVED** [review]

File: apps/web/src/app/admin-console/organizations/policies/password-generator.component.ts [hunk 2]
```diff
@@ -19,31 +22,87 @@ export class PasswordGeneratorPolicy extends BasePolicy {
...
     capitalize: [null],
     includeNumber: [null],
   });
 
-  defaultTypes: { name: string; value: string }[];
+  overridePasswordTypeOptions: { name: string; value: string }[];
+
+  // These subjects cache visibility of the sub-options for passwords
+  // and passphrases; without them policy controls don't show up at all.
+  private showPasswordPolicies = new BehaviorSubject<boolean>(true);
```
Comment (vincentsalucci) [inline apps/web/src/app/admin-console/organizations/policies/password-generator.component.ts:66]: 💭 Do you happen to know if we're still in the practice of using `_` as a prefix for private variables not directly accessed by those consuming this as an observable (below)? Non-blocking of course, it was just my initial reaction!

**Review by vincentsalucci — APPROVED** [review]
Just one potential discussion. Non-blocking, however! 👍

File: apps/web/src/app/admin-console/organizations/policies/password-generator.component.ts (hunk 2 above)
Comment (audreyality) [reply apps/web/src/app/admin-console/organizations/policies/password-generator.component.ts:66]: My guess is that if that were true, we'd lint it.

**Pull request merged**